### PR TITLE
Support mount by PARTUUID

### PR DESCRIPTION
**Pull request by schaefi**
In addition to by-label and by-uuid also support mounting
by PARTUUID. Please note kiwi also makes sure that the grub
generated config file uses the root=PARTUUID= notation and it's
not clear if grub-mkconfig will persist making use of it.
Nevertheless there are also systems which uses different
methods to boot and it makes sense to support partuuid
mappings as well

### CODE REVIEW

**Comment by dcermak** [comment]
According to the grub docs:
```texi
@item GRUB_DISABLE_LINUX_PARTUUID
If @command{grub-mkconfig} cannot identify the root filesystem via its
universally-unique indentifier (UUID), @command{grub-mkconfig} can use the UUID
of the partition containing the filesystem to identify the root filesystem to
the Linux kernel via a @samp{root=PARTUUID=...} kernel parameter.  This is not
as reliable as using the filesystem UUID, but is more reliable than using the
Linux device names. When @samp{GRUB_DISABLE_LINUX_PARTUUID} is set to
@samp{false}, the Linux kernel version must be 2.6.37 (3.10 for systems using
the MSDOS partition scheme) or newer.  This option defaults to @samp{true}.  To
enable the use of partition UUIDs, set this option to @samp{false}.
```

So it looks like you probably want to set `GRUB_DISABLE_LINUX_PARTUUID` to false and then it *should* work :crossed_fingers:. But that sounds like something which we should probably create an integration test for :thinking: ?

**Comment by schaefi** [comment]
@dcermak very good point, thanks. I added a commit to address this

**Comment by schaefi** [comment]
+1 for the integration test. I'll add, not sure when I get to it, thus keeping this one open as a WIP

**Review by davidcassany — APPROVED** [review]
Looks good to me, I agree there should be a simple integration test for that. Gurb features/changes are hard to track, so if we have a tests there are greater chances that it does not silently break over time.

**Review by Conan-Kudo — APPROVED** [review]
LGTM, but we need an integration test for this.

**Comment by schaefi** [comment]
> LGTM, but we need an integration test for this.

https://build.opensuse.org/package/show/Virtualization:Appliances:Images:Testing_x86:leap/test-image-embedded

**Comment by schaefi** [comment]
I placed the test as a profile into the existing embedded testing. I plan to have more features more or less only needed in that space to be tested there

**Comment by schaefi** [comment]
I added a change needed on the dracut module code to support partuuid references. Going to test the integration test and respond back once it all works. Until then, please don't merge, label is set

**Comment by schaefi** [comment]
During testing I found that by-partuuid should be limited to partition tables that actually supports UUIDs natively. I don't want to trust the UUIDs created by Linux from the disk signature. This value can change if you just re-install MBR code (grub2-install) and we don't want to base the boot of a system on a value that can change without even having touched the partition table itself :) Thus I added a runtime check for it

**Comment by schaefi** [comment]
@dcermak The integration is ready and should also be usable within openQA. I tested it locally in QEMU, would be great if you can give it a try too:

* https://build.opensuse.org/package/binaries/Virtualization:Appliances:Images:Testing_x86:leap/test-image-embedded:SystemFeatures/images

after booting you should see by-partuuid device names used in e.g ```/etc/fstab```

Thanks

**Pull request merged**
